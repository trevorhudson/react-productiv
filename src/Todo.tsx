import React from "react";


interface TodoInterface {
  id: string,
  title: string,
  description: string,
  priority: string
}

interface PropsInterface  {
  todo: TodoInterface | null
}

/** Simple presentation component for a todo.
 *
 * Props:
 * - todo: like { id, title, description, priority }
 *
 * { EditableTodo, TopTodo } -> Todo
 **/


const Todo: React.FC<PropsInterface> = ({todo}) =>{
  return (
    <div className="Todo">
      <div><b>{todo!.title}</b> <small>{todo!.priority}</small></div>
      <div><small>{todo!.description}</small></div>
    </div>
  );
}

export default Todo;
