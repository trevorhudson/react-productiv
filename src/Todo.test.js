import { render } from "@testing-library/react";
import Todo from './Todo.js';


describe('Todo Component tests', function () {
  const testTodo = {
    id: 0,
    title: 'title',
    description: 'description',
    priority: 1
  };

  it('renders without crashing', function () {
    render(<Todo todo={testTodo} />);
  });

  it('passes snapshot test', function () {
    const { container } = render(<Todo todo={testTodo} />);
    expect(container).toMatchSnapshot();

  });

});
