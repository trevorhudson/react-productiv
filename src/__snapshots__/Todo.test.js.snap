// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo Component tests passes snapshot test 1`] = `
<div>
  <div
    class="Todo"
  >
    <div>
      <b>
        title
      </b>
       
      <small>
        1
      </small>
    </div>
    <div>
      <small>
        description
      </small>
    </div>
  </div>
</div>
`;
